$NetBSD: patch-Makefile,v 1.5 2022/01/19 02:52:24 pho Exp $

Do not hard-code -ldl, -pthread, or -fPIE as they are handled by the pkgsrc
infrastructure.

--- Makefile.orig	2021-12-31 05:10:03.000000000 +0000
+++ Makefile
@@ -19,7 +19,7 @@ STRIP ?= strip
 OS ?= $(shell uname -s)
 
 # Used for both C and C++
-COMMON_FLAGS = -pthread -fPIE -fno-unwind-tables -fno-asynchronous-unwind-tables
+COMMON_FLAGS = -fno-unwind-tables -fno-asynchronous-unwind-tables
 
 CFLAGS ?= -O2
 CFLAGS += $(COMMON_FLAGS)
@@ -27,7 +27,7 @@ CFLAGS += $(COMMON_FLAGS)
 CXXFLAGS ?= -O2
 CXXFLAGS += $(COMMON_FLAGS) -std=c++20 -fno-exceptions
 CPPFLAGS += -DMOLD_VERSION=\"1.0.1\" -DLIBDIR="\"$(LIBDIR)\""
-LIBS = -pthread -lz -ldl -lm
+LIBS = -lz -lm
 
 SRCS=$(wildcard *.cc elf/*.cc macho/*.cc)
 HEADERS=$(wildcard *.h elf/*.h macho/*.h)
@@ -120,7 +120,7 @@ mold: $(OBJS) $(MIMALLOC_LIB) $(TBB_LIB)
 	ln -sf mold ld64.mold
 
 mold-wrapper.so: elf/mold-wrapper.c Makefile
-	$(CC) $(CPPFLAGS) $(CFLAGS) -fPIC -shared -o $@ $(LDFLAGS) $< -ldl
+	$(CC) $(CPPFLAGS) $(CFLAGS) -fPIC -shared -o $@ $(LDFLAGS) $<
 
 out/%.o: %.cc $(HEADERS) Makefile out/elf/.keep out/macho/.keep
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c -o $@ $<
